plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -0,0 +1,228 @@
+        )
+        if all(isinstance(c, int) for c in (a, b)):
+            scalar.set_integer_casting()
+        return scalar
+    return convert_to_deduced_type(string)  # constant
+
+
+class NevergradSweeper(Sweeper):
+    """Returns a Nevergrad parameter from a definition string.
+
+    Parameters
+    ----------
+    optimizer: str
+       name of a Nevergrad optimizer to use. Some interesting options are:
+         - "OnePlusOne" extremely simple and robust, especially at low budget, but
+           tends to converge early.
+         - "CMA" very good algorithm, but may require a significant budget (> 120)
+         - "TwoPointsDE": an algorithm good in a wide range of settings, for significant budgets
+           (> 120).
+         - "Shiva" an algorithm aiming at identifying the best optimizer given your input
+           definition (work in progress, it may still be ill-suited for low budget)
+       See nevergrad documentation: https://facebookresearch.github.io/nevergrad/
+    budget: int
+       the total number of function evaluation that can be performed
+    num_workers: int
+       the number of evaluation to run in parallel. Sequential means num_workers=1.
+       Population based algorithms such as CMA and DE can have num_workers up to 40 without slowing
+       down the convergence, while OnePlusOne can benefit sequential, but will perform well with several
+       workers as well.
+    noisy: bool
+       notifies (some) algorithms that the function evaluation is noisy
+    maximize: bool
+       whether to perform maximization instead of default minimization
+    seed: int
+        seed for the optimizer for reproducibility. If the seed is -1, the optimizer is not seeded (default)
+    version: int
+       the version of the commandline input parsing. The parsing will probably evolve in the near
+       future and several versions may temporarily coexist.
+       s
+    """
+
+    def __init__(
+        self,
+        optimizer: str,
+        budget: int,
+        num_workers: int,
+        noisy: bool,
+        maximize: bool,
+        seed: Optional[int],
+        version: int,
+    ):
+        self.config: Optional[DictConfig] = None
+        self.launcher: Optional[Launcher] = None
+        assert version == 1, "Only version 1 of API is currently available"
+        self.job_results = None
+        self.optimizer = optimizer
+        self.noisy = noisy
+        self.budget = budget
+        self.num_workers = num_workers
+        self.seed = seed
+        self._direction = -1 if maximize else 1
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.launcher = Plugins.instantiate_launcher(
+            config=config, config_loader=config_loader, task_function=task_function
+        )
+
+    def sweep(self, arguments: List[str]) -> None:
```

Comment:
Is that the API?
I've copied the `ax` one and create a file with the best parameters

Reply:
Which API are you referring to? this is a bit out of context so I am not sure what you are talking about.

Reply:
I meant "what is `sweep` supposed to return? From what I have seen in ax it returns nothing but prints the best parameters in a file

Reply:
Sweep is not returning anything because it's not used as a function by users.
It's always being used from the command line.
If a use case emerges where we want to use the plugin directly we can add a return value and standardize on it when we do.

makes sense?

Reply:
Yep it does, but the example plugin does not follow this API then since it returns a list of batches, hence my confusion

Reply:
Yeah, that's more of wishful thinking. it's not really doing anything right now. thanks for pointing it out.